generator/dumper.cpp

```diff
@@ -17,6 +17,48 @@
 #include "std/queue.hpp"
 #include "std/vector.hpp"
 
+namespace
+{
+template <typename TValue>
+struct SearchTokensCollector
+{
+  priority_queue<pair<uint32_t, strings::UniString>> tokens;
+  strings::UniString m_currentS;
+  uint32_t m_currentCount;
+
+  SearchTokensCollector() : m_currentS(), m_currentCount(0) {}
+
+  void operator()(strings::UniString const & s, TValue const & /* value */)
+  {
+    if (m_currentS == s)
+    {
+      ++m_currentCount;
+    }
+    else
+    {
+      if (m_currentCount > 0)
+      {
+        tokens.push(make_pair(m_currentCount, m_currentS));
+        if (tokens.size() > 100)
```

Comment:
Actually this code is quite bizarre - could you please add a comment describing what this class does? And why not to use "normal" priority queue or even map?